File: Interpreter/PHP/OrExpression.php
<?php

class OrExpression implements IExpression {
    private IExpression $expression1, $expression2;

    public function __construct(IExpression $expression1, IExpression $expression2)
    {
        $this->expression1 = $expression1;
        $this->expression2 = $expression2;
    }

    public function solve(string $context) {
        if (strpos($context, "||") === false) return false;

        return $this->expression1->solve($context) || $this->expression2->solve($context);
    }
}

File: Interpreter/Python 3/OrExpression.py
from IExpression import IExpression

class OrExpression(IExpression):
  def __init__(self, expression1, expression2):
    super().__init__()
    self.__expression1 = expression1
    self.__expression2 = expression2
  
  def solve(self, context):
    if "||" not in context:
      return False

    return self.__expression1.solve(context) or self.__expression2.solve(context)
